src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs

```diff
@@ -27,18 +25,20 @@ public static class WebAssemblyHotReload
 
         internal static async Task InitializeAsync()
         {
-            // Determine if we're running under a hot reload environment (e.g. dotnet-watch).
-            // It's insufficient to know it the app can be hot reloaded (HotReloadFeature.IsSupported),
-            // since the hot-reload agent might be unavailable.
-            if (Environment.GetEnvironmentVariable("DOTNET_MODIFIABLE_ASSEMBLIES") != "debug")
-            {
-                return;
-            }
-
             _hotReloadAgent = new HotReloadAgent(m => Debug.WriteLine(m));
 
-            var jsObjectReference = (IJSUnmarshalledObjectReference)(await DefaultWebAssemblyJSRuntime.Instance.InvokeAsync<IJSObjectReference>("import", "./_framework/blazor-hotreload.js"));
-            await jsObjectReference.InvokeUnmarshalled<Task<int>>("receiveHotReload");
+            try
+            {
+                // Attempt to read previously applied hot reload deltas. dotnet-watch and VS will serve the script that can provide results from local-storage and
+                // the injected middleware if present.
+                var jsObjectReference = (IJSUnmarshalledObjectReference)(await DefaultWebAssemblyJSRuntime.Instance.InvokeAsync<IJSObjectReference>("import", "./_framework/blazor-hotreload.js"));
+                await jsObjectReference.InvokeUnmarshalled<Task<int>>("receiveHotReload");
+            }
+            catch (Exception ex)
+            {
+                // The middleware may be unavailable in some cases - https://github.com/dotnet/aspnetcore/issues/34962. This is not a reason to fail, so swallow the exception and continue.
+                Debug.WriteLine(ex);
+            }
```

Comment:
Is `./_framework/blazor-hotreload.js` served from a middleware?

Is this something we can consider doing during build time? (Possibly automatically loading it when it applies)

Reply:
The reason I'm asking is because users will still see the 404 during development when they open the F12 dev tools.